### PR TITLE
NAS-120608 / 23.10 / fix alert being generated on user caused failover

### PR DESCRIPTION
**Pull request by yocalebo**
The following scenario caused the alert to be generated during QE test cycle.
1. controller A is master controller
2. user initiates a failover via the webUI
3. controller B takes over as master and generates this alert

This alert shouldn't be generated since, almost always, a failover means a controller has been rebooted. This changes the logic just a bit to only generate this alert if we fail to sync the database after 20mins since some of the hardware platforms we sell can take roughly that amount of time to boot up.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/failover_/datastore.py [hunk 1]
```diff
@@ -37,17 +37,14 @@ def set_failure(self):
         try:
             self.send()
         except Exception:
-            self.logger.warning(
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/failover_/datastore.py:40]: Why remove this logging message? (But I think we should remove the traceback though)

File: src/middlewared/middlewared/plugins/failover_/datastore.py [hunk 2]
```diff
@@ -37,17 +37,14 @@ def set_failure(self):
...
-            self.middleware.call_sync('alert.oneshot_create', 'FailoverSyncFailed', None)
-
             def send_retry():
                 set_thread_name('failover_datastore')
 
-                while True:
-                    time.sleep(60)
+                max_time_to_retry = 1200  # 20mins (some platforms take 15-20mins to reboot....)
+                sleep_time = 30
+                while max_time_to_retry > 0:
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/failover_/datastore.py:45]: This converts an infinite loop to infinite, is this change intended?

File: src/middlewared/middlewared/plugins/failover_/datastore.py (hunk 1 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover_/datastore.py:40]: Agree, I added it back (without traceback)

File: src/middlewared/middlewared/plugins/failover_/datastore.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover_/datastore.py:47]: Yeah, I believe that's okay? If we fail to sync db after 20mins, I don't see a reason to continue on, right? The user will be alerted to try and sync the db manually and if it fails again, this code will get started again and repeat the process?

[the remark this answers is not in the document]
Comment (themylogin) [reply src/middlewared/middlewared/plugins/failover_/datastore.py:47]: It has sense to continue the automatic retry infinitely, it is cheap enough. I see no reason why would someone want to trigger manual sync after an extensive period of other node being offline. At least we should allow greater threshold for the boot time

File: src/middlewared/middlewared/plugins/failover_/datastore.py [hunk 3]
```diff
@@ -37,26 +38,24 @@ def set_failure(self):
         try:
             self.send()
         except Exception:
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/failover_/datastore.py:40]:
```suggestion
        except Exception as e:
```

File: src/middlewared/middlewared/plugins/failover_/datastore.py [hunk 4]
```diff
@@ -65,6 +64,11 @@ def send_retry():
                     except Exception:
                         pass
 
+                    if raise_alert_time <= 0 and self.failure:
+                        self.middleware.call_sync('alert.oneshot_delete', 'FailoverSyncFailed', None)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/failover_/datastore.py:68]: This will delete/remove alert every `RAISE_ALERT_SYNC_RETRY_TIME` seconds which will lead to e-mails being sent. Is this intended?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover_/datastore.py:68]: No it's not, I was just trying to avoid opening a duplicate alert if one had already been generated

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover_/datastore.py:68]: Hmm, maybe an email alert every 20mins is actually decent.

**Review by themylogin — APPROVED** [review]
